library/src/main/java/org/eclipse/uprotocol/common/util/log/Log.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.uprotocol.common.util.log;
+
+import static org.eclipse.uprotocol.common.util.UStatusUtils.isOk;
+
+import androidx.annotation.NonNull;
+
+import org.eclipse.uprotocol.v1.UCode;
+import org.eclipse.uprotocol.v1.UStatus;
+
+public final class Log {
```

Comment:
This will cause conflict with uEs that are already using android Log hence forcing them to prefix it with the package name. In addition, many uEs at GM are currently using a custom Logger. Can we either make this internal or can we let the app provide a Logger at initialization?

Reply:
We need formatter mainly for protobuf messages, because protoMessage.toString() dumps everything. Nobody is forced to use this implementation for logging.

Reply:
I have removed Log and Tag. Formatter is enough for logging...